markdown-mode.el

```diff
@@ -9540,6 +9554,8 @@ rows and columns and the column alignment."
   (add-hook 'edit-indirect-after-commit-functions
             #'markdown--edit-indirect-after-commit-function
             nil 'local)
+  (add-hook 'edit-indirect-before-commit-hook
+            #'markdown--edit-indirect-before-commit-hook)
```

Comment:
Can this hook be declared as local hook ?

Reply:
Not really, since this hook needs to be used by the `edit-indirect` buffer that we create. Maybe we could stick this in the `with-current-buffer` and have it be local there?

Reply:
Ah, I see. I try to declare it as local hook then `markdown-edit-code-block` does not work. Thanks for explanation.

Reply:
> Maybe we could stick this in the `with-current-buffer` and have it be local there?

That's probably the best place to set it. It is meant to be entirely local to the newly created edit-indirect buffer, so that's where it should be set IMO.

Reply:
> That's probably the best place to set it. It is meant to be entirely local to the newly created edit-indirect buffer, so that's where it should be set IMO.

sounds good

Reply:
Hey guys, running into an issue here where doing `edit-indirect-save` adds the indent into the edit-indirect buffer. Do you guys know of a way to smuggle the edit-indirect buffer-local definitions into `edit-indirect-after-commit-function`?

Reply:
Just implemented & tested, not many real edge cases. I'll resolve the threads related to the old implementation.